### PR TITLE
gui: upgrade to PySide6 (bug 1782252)

### PR DESCRIPTION
**Pull request by zzzeid**
- upgrade to PySide6
- various renamed methods related to PySide6
- fix deprecated method calls
- upgrade CI images to use ubuntu 20.04
- remove Python 3.6 support

NOTE: This pull request would target a pre-release of version `5.0.0rc0`.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -13,7 +13,7 @@ on:
 
 jobs:
   lint:
-    runs-on: ubuntu-18.04
+    runs-on: ubuntu-20.04
```
Comment (jmaher) [inline .github/workflows/build.yml:16]: would 22.04 be an option?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -39,10 +39,14 @@ jobs:
     strategy:
       matrix:
         os:
-          - ubuntu-18.04
-        python-version: ["3.6", "3.7", "3.8", "3.9", "3.10"]
+          - ubuntu-20.04
```
Comment (jmaher) [inline .github/workflows/build.yml:42]: same here, 22.04 maybe?

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -52,16 +56,16 @@ jobs:
...
+          python -m pip install -e .
       - name: Test
         run: |
           source env/bin/activate
           coverage run -m pytest tests
           coveralls --service=github
 
   build-and-test-linux-gui:
-    runs-on: ubuntu-18.04
+    runs-on: ubuntu-20.04
```
Comment (jmaher) [inline .github/workflows/build.yml:68]: more 20.04 -> 22.04 possibilities

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -70,7 +74,8 @@ jobs:
       - uses: actions/checkout@v2
       - name: setup
         run: |
-          sudo apt install libxkbcommon-x11-0 libxcb-icccm4 libxcb-image0 libxcb-keysyms1 libxcb-randr0 libxcb-render-util0 libxcb-xinerama0 libxcb-xfixes0
+          sudo apt-get update
+          sudo apt install libxkbcommon-x11-0 libxcb-icccm4 libxcb-image0 libxcb-keysyms1 libxcb-randr0 libxcb-render-util0 libxcb-xinerama0 libxcb-xfixes0 libegl-dev -y
```
Comment (jmaher) [inline .github/workflows/build.yml:78]: why -dev?

**Review by jmaher — APPROVED** [review]

File: .github/workflows/build.yml (hunk 4 above)
Comment (zzzeid) [reply .github/workflows/build.yml:78]: This fixes an import error related to Qt:
```
ImportError: libEGL.so.1: cannot open shared object file: No such file or directory
```

File: .github/workflows/build.yml (hunk 1 above)
Comment (zzzeid) [reply .github/workflows/build.yml:16]: Sorry, just realized you made this comment on `lint`, and there's no reason not to use `ubuntu-latest` here. As for the one below (with the matrix), we can probably just add `ubuntu-22.04` to the matrix. GUI tests can probably use a matrix as well, maybe with a couple of versions of Python as well.

I filed [this bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1785095) to track this.

**Review by jmaher — APPROVED** [review]
thanks!

**Review by cgsheeh — APPROVED** [review]
LGTM!

**Pull request merged**